commands/channel.go

```diff
@@ -317,38 +319,48 @@ func getAllDeletedChannelsForTeam(c client.Client, teamID string) ([]*model.Chan
 
 func listChannelsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	teams := getTeamsFromTeamArgs(c, args)
+
+	var multerr *multierror.Error
 	for i, team := range teams {
 		if team == nil {
-			printer.PrintError("Unable to find team '" + args[i] + "'")
+			errMsg := "Unable to find team '" + args[i] + "'"
+			printer.PrintError(errMsg)
+			multerr = multierror.Append(multerr, errors.New(errMsg))
 			continue
 		}
 
 		publicChannels, err := getAllPublicChannelsForTeam(c, team.Id)
 		if err != nil {
-			printer.PrintError(fmt.Sprintf("unable to list public channels for %q: %s", args[i], err))
+			errMsg := fmt.Sprintf("unable to list public channels for %q: %s", args[i], err)
+			printer.PrintError(errMsg)
+			multerr = multierror.Append(multerr, errors.New(errMsg))
 		}
 		for _, channel := range publicChannels {
 			printer.PrintT("{{.Name}}", channel)
 		}
 
 		deletedChannels, err := getAllDeletedChannelsForTeam(c, team.Id)
 		if err != nil {
-			printer.PrintError(fmt.Sprintf("unable to list archived channels for %q: %s", args[i], err))
+			errMsg := fmt.Sprintf("unable to list archived channels for %q: %s", args[i], err)
+			printer.PrintError(errMsg)
+			multerr = multierror.Append(multerr, errors.New(errMsg))
 		}
 		for _, channel := range deletedChannels {
 			printer.PrintT("{{.Name}} (archived)", channel)
 		}
 
 		privateChannels, appErr := getPrivateChannels(c, team.Id)
 		if appErr != nil {
-			printer.PrintError(fmt.Sprintf("unable to list private channels for %q: %s", args[i], appErr.Error()))
+			errMsg := fmt.Sprintf("unable to list private channels for %q: %s", args[i], appErr.Error())
+			printer.PrintError(errMsg)
+			multerr = multierror.Append(multerr, errors.New(errMsg))
```

Comment:
We are not canonically wrapping the error, I think we can just pass the `appError` to the `multerr`, or do a proper wrapping with `%w` while formatting.